websec.1
.\" 
.TH "WEBSEC" "1" "1.4.0" "" ""
.SH "NAME"
websec \- web secretary

.SH "SYNOPSIS"
\fBwebsec\fR <$HOME/.websec/url.list>

.SH "DESCRIPTION"
\fBWebsec\fR is a web page monitoring software. 
It will send you a changed web page with the contents
highlighted. The debian version of websec has been
changed, so the web\-pages archive will be stored in
\fI$HOME/.websec\fR, this is also the directory where
you should keep your url.list.


When called without an argument, \fBwebsec\fR will try to
open \fI$HOME/.websec/url.list\fR, but you can also 
provide a filename as the first argument.


You can add a line like \fIAddSubject = [websec]\fR to
url.list, websec will add \fI[websec]\fR to every subject
as a first word when mail is sent. You can then easily detect
this line by a mail filter.


The keywords \fIRetry\fR, \fIRetrywait\fR, and
\fITimeout\fR in url.list lets you specify the number
of times to retry, time to wait between retries, and a
timeout setting.


\fBWebsec\fR waits for a random number of seconds between
retries up to the value specified by the \fIRandomwait\fR
keyword. This is to prevent websec from being blocked by
websites that perform log analysis to find time similarities
between requests.



.SH "SEE ALSO"
/usr/share/doc/websec/README.gz.

.SH "AUTHOR"
This manual page was written by Joop Stakenborg
<pa3aba@debian.org>,
for the Debian GNU/Linux system (but may be used by others).


NEWS
1.7 - Under construction

    * Add MailFrom, to set the address from which the mail will appear to come.
      (Thanks to Adam Stanley)

    * Added URL to the mail subject, to simplify jumping to a changed page from
      the mail folder.

    * Added an EmailError option to disable sending error messages. This was
      requested numerous times (Bug#3498 on Savannah), this fix is based on a
      patch from Peter Bieringer.

    * Add man pages for url.list and ignore.list, removed their documentation
      from the README file and moved it to the files themselves for easier
      reference.

    * Add "ascii" color highlighting from Javier M. Mora. Instead of using color
      to highlight it uses ascii characters around the change.

    * Add Emacs mode to highlight url.list, it is likely to be in error, a user
      help with it would be most helpful as I'm not an Emacs user.

    * Add ViM syntax file to highlight url.list.

    * Fixed usage of environment variables in definitions, also allowed using
      multiple environment variables in a single line.

1.6.0 - Released on 05 May 2003

    * Added DateFMT feature to set Date Formats in mail subjects.

    * Add to EmailLink the local file system link so that a user can open it in
      his browser.

    * Add Program option to open a program on changed link instead of mailing
      notification.

    * Multiple bug fixes.

1.5.0 - Released on 02 May 2003

    * Added UserAgent option to control the user agent sent by the web client.
      This enables access to websites which lockdown unknown browsers.

    * Enabled the user to run websec from the current directory without the
      multi-user ~/.websec/ setup, the multi-user setup is the default unless
      you run websec from a directory with the url.list file.

    * Allow using environment variables in config file, this enables things
      like: Email = ${USER}@example.com

    * Moved the manpages to pod and into the files, this enables fancy options
      such as -help and -man to work and reduces duplication of effort.

1.4.0 - Released on 31 Oct 2002

    * Various changes that came from the Debian package

1.3.4 - Released on 6 Jun 2000

    * Added tip on how to install LWP in READE. Thanks to Jeff for contributing
      this info!

    * Fixed a bug with $emailLink and added digest format. Thanks to Matti Airas
      for contributing this!

1.3.3 - Released on 31 Jan 2000

    * Added MIME-Version header to email sent. Thanks to Joe Rumsey for pointing
      this out. This fixes problems under certain mailers, where messages are
      treated as plain text instead of HTML.

1.3.2 - Released on 2 Dec 1999

    * Added <CODE> as a recognized tag.

    * Added the ability to send updated page to multiple recipients.

    * Added the ability to send URL link instead of entire page to recipients.
      Check "EmailLink" tag.

    * "Alexander Lazic" <al@eunet.at> suggested using strftime instead of the
      Unix command "date". This has been fixed.

1.31 - Released on 17 Apr 1999

    * Volker Stampa contributed some code to allow websec to work with proxies
      that require authentication.

1.3  - Released on 20 Mar 1999

    * Trevor Boicey suggested allowing the use of arbitrary HTML colors in the
      "-hicolor" parameter of webdiff. This feature has been included.

    * Webdiff had some problems with a tag of this nature:
      <A HREF="xxx <yyy>">, first found in the ZDNET series of web sites.  This
      has been fixed.

    * A new "ignore URL" feature has been included. This allows certain
      hyperlinks sections in a web page to be skipped during webdiff
      processing.

    * All ignore keywords and URLs have been consolidated into one file.

1.22 - Released on 13 Jan 1999

    * A small shell script has been included to "rollback" the files in the
      archive directory for one session.

    * Proxy settings can now be supplied via the "http_proxy" environment
      variable. However, the "Proxy" parameter will take precedence over the
      environment variable.

    * When checking for short and long tokens (based on the Tmin and Tmax
      parameters), any mangled HTML tags are first stripped from the token
      before word count is done.  Therefore, word count is done on the "plain
      text" version of the token.

    * When checking for ignore keywords, the token which possibly contains
      mangled HTML tags is first checked. Then it is stripped of any mangled
      HTML tags and checked again. This is cater for cases where the mangled
      HTML tag precedes or follows an actual word without any spacing. Hence
      the entire string is treated as one word, and will fail to match any of
      the ignore keywords.

1.21 - Released on 1 Jan 1999 (Happy New Year!)

    * Made minor modification to try downloading any URL up to 3 times before
      giving up. I did not find it necessary to include this as a parameter, so
      it was hard-coded.

1.2  - Released on 25 Dec 1998 (Merry Christmas!)

    * Rewrote Web Secretary to use the LWP module in Perl for HTTP retrieval and
      email transmission. Hence, it is no longer necessary to have 'lynx' and
      'metasend' installed on your system in order to use Web Secretary.

    * Since Web Secretary was rewritten to use the LWP module (instead of lynx),
      for HTTP retrieval, I had to add a 'Proxy' parameter for folks (like
      myself) who are behind a firewall.

    * Added the 'Tmin' parameter to ignore short tokens when highlighting
      differences. This is useful because certain sites have tokens containing
      one or two words which change constantly but are uninteresting to track.

    * Added the 'Tmax' parameter to prevent long tokens from being processed by
      the ignore keywords filter. This was included because certain sites have
      tokens containing the current day/month etc. which I want to filter off.

websec.el
;;; url-list-mode.el --- simple mode for Web Secretary url files ("url.list")

;; Copyright (C) 2002 Christoph Conrad

;; Author: Christoph Conrad <christoph.conrad@gmx.de>
;; Created: 11 Feb 2002
;; Version: 0.5
;; Keywords: fontlock web secretary

;; This program is free software; you can redistribute it and/or
;; modify it under the terms of the GNU General Public License as
;; published by the Free Software Foundation; either version 2, or (at
;; your option) any later version.

;; This program is distributed in the hope that it will be useful, but
;; WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
;; General Public License for more details.

;; You should have received a copy of the GNU General Public License
;; along with GNU Emacs; see the file COPYING. If not, write to the
;; Free Software Foundation, Inc., 59 Temple Place - Suite 330,
;; Boston, MA 02111-1307, USA.

;; Not available in XEmacs
(require 'generic)
(require 'font-lock)

;; Keywords embedded anywhere (except [normally] in quoted strings)
(defvar url-list-font-lock-list
  '(("^\\([^=\n\r]*\\)=\\([^\n\r]*\\)$"
     (2 font-lock-variable-name-face))))

(defvar url-list-keywords
  '( "URL" "Auth" "Name" "Prefix" "Diff" "Hicolor" "Ignore" "IgnoreURL"
     "Tmin" "Tmax" "Proxy" "ProxyAuth" "Email" "EmailLink" "Digest" ))

(define-generic-mode 'url-list-mode
  ;; comment-list
  '("#")
  ;; keyword-list
  url-list-keywords
  ;; font-lock-list
  url-list-font-lock-list
  ;; auto-mode-list
  '("/url\.list$")
  ;; function-list for setup
  nil
  "Mode for highlighting Web Secretary URL files.")

(provide 'url-list-mode)
